internal/temporalcli/commands.extension.go

```diff
@@ -0,0 +1,74 @@
+package temporalcli
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+const (
+	// extensionPrefix is the required prefix for all Temporal CLI extensions.
+	extensionPrefix = "temporal-"
+)
+
+// lookupExtension searches for an extension executable from the given args.
+// Extension commands must come before any flags.
```

Comment:
This still is an active research question; but in order to move forward and gain more data we make a decision here. This can/will be revisited.